junit-platform-commons/src/main/java/org/junit/platform/commons/util/ClassFilter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015-2017 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.commons.util;
+
+import static org.apiguardian.api.API.Status.INTERNAL;
+
+import java.util.function.Predicate;
+
+import org.apiguardian.api.API;
+
+/**
+ * Class-related predicate holder used by reflection utilities.
+ *
+ * <h3>DISCLAIMER</h3>
+ *
+ * <p>These utilities are intended solely for usage within the JUnit framework
+ * itself. <strong>Any usage by external parties is not supported.</strong>
+ * Use at your own risk!
+ *
+ * @since 1.1
+ */
+@API(status = INTERNAL, since = "1.1")
+public class ClassFilter implements Predicate<Class<?>> {
+
+	public static ClassFilter of(Predicate<Class<?>> classPredicate) {
+		return of(name -> true, classPredicate);
+	}
+
+	public static ClassFilter of(Predicate<String> namePredicate, Predicate<Class<?>> classPredicate) {
+		Preconditions.notNull(namePredicate, "name predicate must not be null");
+		Preconditions.notNull(classPredicate, "class predicate must not be null");
```

Comment:
I think these precondition checks should be moved to the constructor.

Reply:
Moved via https://github.com/junit-
team/junit5/pull/1078/commits/1a6623b6d27595ebbc5105a869554ca8d896298a